src/dataEntities/block.ts

```diff
@@ -50,3 +54,87 @@ export interface ResponderBlock extends IBlockStub {
 }
 
 export interface Block extends IBlockStub, Logs, Transactions, TransactionHashes {}
+
+export class BlockItemStore<TBlock extends IBlockStub> extends StartStopService {
+    private readonly subDb: LevelUp<EncodingDown<string, any>>;
+    constructor(db: LevelUp<EncodingDown<string, any>>) {
+        super("block-item-store");
+        this.subDb = sub(db, `block-item-store`, { valueEncoding: "json" });
+    }
+
+    private itemsByHeight: Map<number, Set<string>> = new Map();
+    private items: Map<string, any> = new Map();
+
+    protected async startInternal() {
+        // load all items from the db
+        for await (const record of this.subDb.createReadStream()) {
+            const { key, value } = (record as any) as { key: string; value: any };
+
+            const i = key.indexOf(":");
+            const height = Number.parseInt(key.substring(0, i));
+            const memKey = key.substring(i + 1);
+
+            const itemsAtHeight = this.itemsByHeight.get(height);
+            if (itemsAtHeight) itemsAtHeight.add(memKey);
+            else this.itemsByHeight.set(height, new Set([memKey]));
+            this.items.set(memKey, value);
+        }
+    }
+    protected async stopInternal() {}
+
+    public async putBlockItem(blockHeight: number, blockHash: string, itemKey: string, item: any) {
+        const memKey = `${blockHash}:${itemKey}`;
+        const dbKey = `${blockHeight}:${memKey}`;
+
+        const itemsAtHeight = this.itemsByHeight.get(blockHeight);
+        if (itemsAtHeight) itemsAtHeight.add(memKey);
+        else this.itemsByHeight.set(blockHeight, new Set([memKey]));
+        this.items.set(memKey, item);
+
+        await this.subDb.put(dbKey, item);
+    }
+
+    /**
+     * Gets the item with key `itemKey` for block `blockHash`.
+     * Returns `undefined` if a key is not present.
+     **/
+    public getItem(blockHash: string, itemKey: string) {
+        const key = `${blockHash}:${itemKey}`;
+        return this.items.get(key);
+    }
+
+    // This behavior is too specific for the needs of the BlockCache; should be refactored
```

Comment:
This comment seems out of date, or like it belongs on a ticket.

Reply:
Removed, it might likely be part of #316 anyway.